Automattic/pocket-casts-ios
# Feature - Profile Bookmarks

Fixes #

This PR implements a central bookmarks screen where you can view all your bookmarks across podcasts.
| 1 | 2 | 3 |
| - | - | - |
| ![Bookmarks_Entry](https://github.com/Automattic/pocket-casts-ios/assets/651601/4b3281ca-85e3-46d5-a0b2-07c9e971c044) | ![Bookmarks_empty](https://github.com/Automattic/pocket-casts-ios/assets/651601/c7c241bd-d386-44f8-89fc-94d88545ac1e) | ![Bookmarks_No_Subscription](https://github.com/Automattic/pocket-casts-ios/assets/651601/d231edd8-671e-430c-b2cb-5c1bb0c16402) |
| ![Bookmarks_Search](https://github.com/Automattic/pocket-casts-ios/assets/651601/7de99cf9-af49-44cf-afe1-e391d2c62d9b) | ![Bookmarks_More_Options](https://github.com/Automattic/pocket-casts-ios/assets/651601/a37b436c-fcc3-4f7e-93d8-06ca8ea0347f) | ![Bookmarks_edit](https://github.com/Automattic/pocket-casts-ios/assets/651601/73567a26-e847-4b4f-bdd4-29df6a87afe9) |
| ![Bookmarks_multi_select](https://github.com/Automattic/pocket-casts-ios/assets/651601/8e3bc291-977e-4804-a840-2a3b2ed74953) | ![Bookmarks_Select_For_Delete](https://github.com/Automattic/pocket-casts-ios/assets/651601/0bae20bb-08ac-4443-bcb0-598342156229) | ![Bookmarks_Delete_Confirmation](https://github.com/Automattic/pocket-casts-ios/assets/651601/55558196-b9b8-4185-9c8c-e2e26063c2f0) |

## To test

- Start the app with an user that has Plus or Patron active
- Enable the feature flag for profile bookmarks: Profile -> Settings -> Beta Features - >  profileBookmarks
- Go to profile
- Scroll down and check if you see a Bookmarks option
- Tap on Bookmarks
- Check if the list of your bookmarks is seen ( if you never set a bookmark you should see an empty view that says how to set it)
- Tap on the play of a bookmark and check if it start the bookmark
- Check if sort option works
  - Tap on the more button (...) on the top right
  - Tap on Sort By
  - Switch between different sort orders and check if it works correctly
- Check if delete works
  - Tap on more button again
  - Tap on select bookmarks
  - Select some bookmarks
  - Tap on the delete icon ( trash button)
  - Confirm the deletion
  - Check that the delete happens
- Check if edit works
  - Tap on more button again
  - Tap on select bookmarks
  - Select one bookmark
  - Tap on the edit button
  - Change the name of the bookmark
  - Confirm
  - Check if the name updates correctly
- Check event tracking
  - Enable tracks logging
  - Interact with bookmarks on the profile (change name, delete)
  - Check if tracks events have the `source` parameters set to `profile_screen`  

### Test without Plus
- Start the app with an user without Plus or Patron
- Enable the feature flag for profile bookmarks: Profile -> Settings -> Beta Features - >  profileBookmarks
- Go to profile
- Scroll down and check if you see a Bookmarks option
- Tap on Bookmarks
- Check if it shows a text that allows to redirect to the Plus purchase flow

Please test the Bookmarks screen with the different themes to see if all is displayed correctly

## Checklist
